libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/roadobject/RoadObjectEx.kt

```diff
@@ -0,0 +1,65 @@
+@file: JvmName("RoadObjectEx")
+
+package com.mapbox.navigation.core.trip.roadobject
+
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.eh.EHorizon
+import com.mapbox.navigation.base.trip.model.roadobject.UpcomingRoadObject
+import com.mapbox.navigation.core.trip.roadobject.reststop.RestStopFactory
+import com.mapbox.navigation.core.trip.roadobject.reststop.RestStopFromIntersection
+import com.mapbox.navigation.core.trip.roadobject.reststop.RestStopFromRoadObject
+import com.mapbox.navigation.utils.internal.ifNonNull
+
+/**
+ * Use the API to retrieve a list of all upcoming rest stops using [UpcomingRoadObject].
+ * This is recommended to be used with:
+ * - [EHorizon] during Free Drive or
+ * - [RouteProgress] during Active Navigation.
+ */
+fun List<UpcomingRoadObject>.getAllRestStops(): List<RestStopFromRoadObject> {
+    return RestStopFactory.createWithUpcomingRoadObjects(this)
+}
+
+/**
+ * Use the API to retrieve the first upcoming rest stops using [UpcomingRoadObject].
+ * This is recommended to be used with:
+ * - [EHorizon] during Free Drive or
+ * - [RouteProgress] during Active Navigation.
+ */
+fun List<UpcomingRoadObject>.getFirstUpcomingRoadObject(): RestStopFromRoadObject? {
+    val restStops = this.getAllRestStops()
+    if (restStops.isNotEmpty()) {
+        val distanceToStart = restStops.first().distanceToStart
```

Comment:
should try to find first rest stop with nonNull `distanceToStart`?

Reply:
Maybe even appropriate distanceToStart, not just nonNull? I mean so that the if below returns true.

Reply:
`distanceToStart` has to be `null`. This the documentation of `distanceToStart` from `nav-native`
```
@param distanceToStart remaining distance to the start of the object.
 * This value will be negative after passing the start of the object and until we cross the finish
 * point of the [RoadObject]s geometry for objects that are on the actively navigated route,
 * but it will be zero for [EHorizon] objects. It will be null if couldn't be determined.
```